cmd/anonymizer/app/anonymizer/anonymizer.go

```diff
@@ -88,14 +95,31 @@ func New(mappingFile string, options Options, logger *zap.Logger) *Anonymizer {
 			logger.Fatal("Cannot unmarshal previous mapping", zap.Error(err))
 		}
 	}
+	a.wg.Add(1)
 	go func() {
-		for range time.NewTicker(10 * time.Second).C {
-			a.SaveMapping()
+		defer a.wg.Done()
+		ticker := time.NewTicker(10 * time.Second)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				a.SaveMapping()
+			case <-a.ctx.Done():
+				return
+			}
 		}
 	}()
 	return a
 }
 
+func (a *Anonymizer) WaitForStop() {
```

Comment:
I don't think this needs to be a public function, you just need to call Wait after cancel() in Stop()

Reply:
this way you only return from Stop() once we confirmed that all background work has stopped (even though goroutine itself might linger before exiting)

Reply:
Done!, `Stop()` now calls `wg.Wait()` after cancelling the context to block until the background goroutine exits